File: backend/RaceRoHub/src/main/java/com/racero/hub/model/Car.java
package com.racero.hub.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Car {
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Id
    public Long id;

    public String model;
    public String engine;
    public String suspensions;
    public String gearbox;
    public String brakes;
    public String wheels;
    public String aero;
}

File: frontend/racerohub/lib/models/car.dart
class Car {
  final int? id;
  final String model;
  final String engine;
  final String suspensions;
  final String gearbox;
  final String brakes;
  final String wheels;
  final String aero;

  Car({
    this.id,
    required this.model,
    required this.engine,
    required this.suspensions,
    required this.gearbox,
    required this.brakes,
    required this.wheels,
    required this.aero,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'],
      model: json['model'] ?? '',
      engine: json['engine'] ?? '',
      suspensions: json['suspensions'] ?? '',
      gearbox: json['gearbox'] ?? '',
      brakes: json['brakes'] ?? '',
      wheels: json['wheels'] ?? '',
      aero: json['aero'] ?? '',
    );
  }
}
